src/base/checked_runner.ml

```diff
@@ -341,12 +349,14 @@ struct
           Checked_intf.Basic
             with module Types := Types
             with type field := Checked_runner.field
+             and type run_state := run_state
```

Comment:
Not blocking: wondering if we could not remove some type aliases above to simplify the interface. `type state = run_state` for instance. Only looking at the lines above, it might not make sense as used for something else.